types/V1.mli

```diff
@@ -125,40 +125,38 @@ module type FLOW = sig
       logging. *)
 
   val read: flow -> [`Ok of buffer | `Eof | `Error of error ] io
-  (** [read flow] will block until it either successfully reads a
-      segment of data from the current flow, receives an [Eof]
-      signifying that the connection is now closed, or an [Error]. *)
+  (** [read flow] reads some data from the flow and returns a
+      fresh buffer (of arbitrary size), an [`Eof] if the other side has
+      called [close] or an [`Error]. *)
```

Comment:
Hm good point. Re-reading it again I think when I read the previous "successfully reads a segment of data" I imagine it returning more than a 0 byte segment. Perhaps my comment "(of arbitrary size)" makes it sound like 0 length is acceptable ... quite like non-blocking I/O on Unix? Perhaps I should clarify that the "arbitrary size" must be > 0 bytes which implies that the function must block until at least 1 byte is available?